skyportal/handlers/api/tag.py

```diff
@@ -0,0 +1,238 @@
+            ).first()
+
+            if not tag:
+                return self.error("Tag not found", status=404)
+
+            session.delete(tag)
+            session.commit()
+
+            return self.success(f"Successfully deleted tag {tag}")
+
+
+class ObjTagHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """Get all tag-obj associations or filter by obj_id/objtagoption_id"""
+        obj_id = self.get_query_argument("obj_id", None)
+        objtagoption_id = self.get_query_argument("objtagoption_id", None)
+
+        with self.Session() as session:
+            query = ObjTag.select(session.user_or_token)
+
+            if obj_id:
+                query = query.where(ObjTag.obj_id == obj_id)
+            if objtagoption_id:
+                query = query.where(ObjTag.objtagoption_id == objtagoption_id)
+
+            associations = session.scalars(query).all()
+            return self.success(associations)
+
+    @auth_or_token
+    def post(self):
+        """Create a new tag-obj association"""
+        data = self.get_json()
+        objtagoption_id = data.get("objtagoption_id")
+        obj_id = data.get("obj_id")
+
+        if not objtagoption_id or not obj_id:
+            return self.error("Both `objtagoption_id` and `obj_id` must be provided")
+
+        with self.Session() as session:
+            # Check if association already exists
+            if session.scalars(
+                ObjTag.select(session.user_or_token)
+                .where(ObjTag.objtagoption_id == objtagoption_id)
+                .where(ObjTag.obj_id == obj_id)
+            ).first():
+                return self.error("This tag-obj association already exists")
+
+            # Verify tag exists
+            if not session.scalars(
+                ObjTagOption.select(session.user_or_token).where(
+                    ObjTagOption.id == objtagoption_id
+                )
+            ).first():
+                return self.error("Specified tag does not exist", status=404)
+
+            # Verify obj exists
+            if not session.scalars(
+                Obj.select(session.user_or_token).where(Obj.id == obj_id)
+            ).first():
+                return self.error("Specified obj does not exist", status=404)
+
+            if hasattr(self.current_user, "created_by"):
+                author_id = self.current_user.created_by.id
+            else:
+                author_id = self.current_user.id
+
+            new_assoc = ObjTag(
+                objtagoption_id=objtagoption_id, obj_id=obj_id, author_id=author_id
+            )
+            session.add(new_assoc)
+            session.commit()
+
+            return self.success(new_assoc)
+
+    @auth_or_token
+    def patch(self, association_id):
+        """Update an existing tag-obj association"""
+        data = self.get_json()
+        new_tag_id = data.get("objtagoption_id")
+        new_obj_id = data.get("obj_id")
+
+        if hasattr(self.current_user, "created_by"):
```

Comment:
same comment here about the user id. In fact that comment is valid all across that file.